src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/DotNetHelper.cs

```diff
@@ -187,6 +187,10 @@ public void ExecutePublish(string projectName, bool? selfContained = null, strin
                     options += " /p:PublishReadyToRun=true";
                     binlogDifferentiator += "-R2R";
                 }
+                if (!string.IsNullOrEmpty(customOptions))
```

Comment:
Is there a reason why this is conditioned to only be applicable in selfContained scenarios?  It feels generally applicable.  If it is self contained specific, it should be renamed accordingly.

Reply:
Good point, will move up